src/runtime_strings.js

```diff
@@ -376,9 +376,9 @@ function lengthBytesUTF32(str) {
 
 // Allocate heap space for a JS string, and write it there.
 // It is the responsibility of the caller to free() that memory.
-function allocateUTF8(str) {
+function allocateUTF8(str, allocator) {
```

Comment:
this increases code size, and makes me think - how about an alternative approach, where you do things like this?

```
function runWithBuiltinMalloc(func) {
  var prev = _malloc;
  _malloc = _builtin_malloc;
  try {
    func();
  } finally {
    _malloc = prev;
  }
}
```

This will put the special code in the more obvious place, perhaps, and avoid touching irrelevant code like `allocateUTF8`?

Reply:
Done.